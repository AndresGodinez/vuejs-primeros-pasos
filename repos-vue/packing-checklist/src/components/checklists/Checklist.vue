<template>
  <article class="checklist">
    <h1 class="checklist__title">{{title}}</h1>
    <template v-for="sublist in sublists">
      <sub-checklist :title="sublist.title" :items="sublist.items" :key="sublist.id" />
    </template>
  </article>
</template>

<script>
import SubChecklist from '@/components/checklists/SubChecklist'

export default {
  name: 'checklist',
  props: ['title', 'sublists'],
  components: {
    SubChecklist
  }

}
</script>

<style lang="sass" scoped>
@import "~@/sass/_vars.sass"

.checklist
  margin: 2em 0

.checklist__title
  font-size: 1.2em
  border-bottom: 1px solid $main-color
</style>
