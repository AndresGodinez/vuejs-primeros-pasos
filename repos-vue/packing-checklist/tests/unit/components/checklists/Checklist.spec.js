import { shallow } from '@vue/test-utils'
import Checklist from '@/components/checklists/Checklist.vue'

describe('Checklist', () => {
  it('has the right DOM structure', () => {
    const wrapper = shallow(Checklist, {
      propsData: { title: 'a checklist', sublists: [] }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
