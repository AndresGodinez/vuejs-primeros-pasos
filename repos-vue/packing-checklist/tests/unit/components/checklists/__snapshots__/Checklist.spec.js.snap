// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checklist has the right DOM structure 1`] = `
<article
  class="checklist"
>
  <h1
    class="checklist__title"
  >
    a checklist
  </h1>
   
</article>
`;
